ccl/backupccl: TestOnlineRestoreBasic failed

ccl/backupccl.TestOnlineRestoreBasic [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15838334?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15838334?buildTab=artifacts#/) on master @ [e410fb5c3425f67d3ff8a7968064032acc76f57b](https://github.com/cockroachdb/cockroach/commits/e410fb5c3425f67d3ff8a7968064032acc76f57b):


```
=== RUN   TestOnlineRestoreBasic
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/44cc1cc0ef77f902d6c266f2a388abb1/logTestOnlineRestoreBasic33038696
    test_log_scope.go:81: use -show-logs to present logs inline
    testutils.go:299: no Invalid Descriptors
    testutils.go:299: no Invalid Descriptors
    restore_online_test.go:96: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/44cc1cc0ef77f902d6c266f2a388abb1/logTestOnlineRestoreBasic33038696
--- FAIL: TestOnlineRestoreBasic (4.61s)
=== RUN   TestOnlineRestoreBasic/incremental=true
    restore_online_test.go:90: query 'SELECT create_statement FROM [SHOW CREATE TABLE data.bank]': expected:
        CREATE TABLE public.bank (
        	id INT8 NOT NULL,
        	balance INT8 NULL,
        	payload STRING NULL,
        	CONSTRAINT bank_pkey PRIMARY KEY (id ASC),
        	FAMILY fam_0_id_balance_payload (id, balance, payload)
        )
        
        got:
        CREATE TABLE public.bank (
        	id INT8 NOT NULL,
        	balance INT8 NULL,
        	payload STRING NULL,
        	CONSTRAINT bank_pkey PRIMARY KEY (id ASC),
        	FAMILY fam_0_id_balance_payload (id, balance, payload)
        ) WITH (sql_stats_automatic_collection_enabled = false)
        
    --- FAIL: TestOnlineRestoreBasic/incremental=true (1.37s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestOnlineRestoreBasic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

ccl/backupccl.TestOnlineRestoreBasic [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15989856?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15989856?buildTab=artifacts#/) on master @ [6b22014dec8c5e8e6006d710ed05bab5b01b4ae2](https://github.com/cockroachdb/cockroach/commits/6b22014dec8c5e8e6006d710ed05bab5b01b4ae2):


```
=== RUN   TestOnlineRestoreBasic
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/44cc1cc0ef77f902d6c266f2a388abb1/logTestOnlineRestoreBasic3049318939
    test_log_scope.go:81: use -show-logs to present logs inline
    testutils.go:299: no Invalid Descriptors
    testutils.go:299: no Invalid Descriptors
    restore_online_test.go:96: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/44cc1cc0ef77f902d6c266f2a388abb1/logTestOnlineRestoreBasic3049318939
--- FAIL: TestOnlineRestoreBasic (5.57s)
=== RUN   TestOnlineRestoreBasic/incremental=true
    restore_online_test.go:90: query 'SELECT create_statement FROM [SHOW CREATE TABLE data.bank]': expected:
        CREATE TABLE public.bank (
        	id INT8 NOT NULL,
        	balance INT8 NULL,
        	payload STRING NULL,
        	CONSTRAINT bank_pkey PRIMARY KEY (id ASC),
        	FAMILY fam_0_id_balance_payload (id, balance, payload)
        )
        
        got:
        CREATE TABLE public.bank (
        	id INT8 NOT NULL,
        	balance INT8 NULL,
        	payload STRING NULL,
        	CONSTRAINT bank_pkey PRIMARY KEY (id ASC),
        	FAMILY fam_0_id_balance_payload (id, balance, payload)
        ) WITH (sql_stats_automatic_collection_enabled = false)
        
    --- FAIL: TestOnlineRestoreBasic/incremental=true (1.34s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestOnlineRestoreBasic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/ccl/backupccl/restore_online_test.go
+++ b/pkg/ccl/backupccl/restore_online_test.go
@@ -212,7 +212,7 @@ func TestOnlineRestoreWaitForDownload(t *testing.T) {
 
 	// Wait for the download job to complete.
 	var downloadJobID jobspb.JobID
-	sqlDB.QueryRow(t, `SELECT job_id FROM [SHOW JOBS] WHERE description LIKE '%Background Data Download%'`).Scan(&downloadJobID)
+	sqlDB.QueryRow(t, `SELECT job_id FROM [SHOW JOBS] WHERE description LIKE '%Background Data Download%' ORDER BY created DESC LIMIT 1`).Scan(&downloadJobID)
 	jobutils.WaitForJobToSucceed(t, sqlDB, downloadJobID)
 
 	sqlDB.CheckQueryResults(t, jobutils.GetExternalBytesForConnectedTenant, [][]string{{"0"}})
@@ -376,8 +376,6 @@ func bankOnlineRestore(
 	sqlDB.Exec(t, "CREATE DATABASE data")
 	sqlDB.Exec(t, fmt.Sprintf("RESTORE TABLE data.bank FROM LATEST IN '%s' WITH EXPERIMENTAL DEFERRED COPY", externalStorage))
 
-	require.Equal(t, checkLinkingProgress(t, sqlDB), float32(1.0))
-
 	var restoreRowCount int
 	sqlDB.QueryRow(t, "SELECT count(*) FROM data.bank").Scan(&restoreRowCount)
 	require.Equal(t, numAccounts, restoreRowCount)
@@ -415,11 +413,4 @@ func assertOnlineRestoreWithRekeying(
 	sqlDB.QueryRow(t, bankTableIDQuery).Scan(&originalID)
 	rSQLDB.QueryRow(t, bankTableIDQuery).Scan(&restoreID)
 	require.NotEqual(t, originalID, restoreID)
-}
-
-func checkLinkingProgress(t *testing.T, sqlDB *sqlutils.SQLRunner) float32 {
-	var linkingJobID jobspb.JobID
-	sqlDB.QueryRow(t, `SELECT job_id FROM [SHOW JOBS] WHERE job_type = 'RESTORE' ORDER BY created LIMIT 1`).Scan(&linkingJobID)
-	prog := jobutils.GetJobProgress(t, sqlDB, linkingJobID)
-	return prog.GetFractionCompleted()
 }

